src/data.rs

```diff
@@ -0,0 +1,272 @@
+//! Custom structs and enums for mac-notification-sys.
+
+use crate::utilities::check_sound;
+use objc_foundation::{INSDictionary, INSString, NSDictionary, NSString};
+use objc_id::Id;
+use std::ops::Deref;
+
+/// Possible actions accessible through the main button of the notification
+pub enum MainButton<'a> {
+    /// Display a single action with the given name
+    ///
+    /// # Example:
+    ///
+    /// ```no_run
+    /// # use mac_notification_sys::*;
+    /// let _ = MainButton::SingleAction("Action name");
+    /// ```
+    SingleAction(&'a str),
+    /// Display a dropdown with the given title, with a list of actions with given names
+    ///
+    /// # Example:
+    ///
+    /// ```no_run
+    /// # use mac_notification_sys::*;
+    /// let _ = MainButton::DropdownActions("Dropdown name", &["Action 1", "Action 2"]);
+    /// ```
+    DropdownActions(&'a str, &'a [&'a str]),
+    /// Display a text input field with the given placeholder
+    ///
+    /// # Example:
+    ///
+    /// ```no_run
+    /// # use mac_notification_sys::*;
+    /// let _ = MainButton::Response("Enter some text...");
+    /// ```
+    Response(&'a str),
+}
+
+/// Options to further customize the notification
+pub struct NotificationOptions<'a> {
+    pub(crate) main_button: Option<MainButton<'a>>,
```

Comment:
I get your idea with the skipped builder struct but now clips warns `new_without_default` what is your opinion on this @hoodie? Is this clean enough? Because a default implementation which only calls ::new() would be rather pointless...